datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachActions.java

```diff
@@ -111,6 +115,12 @@ protected void doExecute(SdxContext context, SdxEvent payload, Map<Object, Objec
                         stackV4Endpoint.updateNameAndCrn(0L,  (String) variables.get(SDX_NAME), initiatorUserCrn,
                                 sdxCluster.getClusterName(), sdxCluster.getCrn()));
                 sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.STOPPED, "Data lake detaching in progress", payload.getResourceId());
+                if (variables.containsKey(NEW_SDX)) {
+                    SdxCluster newSdxCluster = (SdxCluster) variables.get(NEW_SDX);
+                    LOGGER.info("Updating the cluster CRN from {} to {}", newSdxCluster.getCrn(), sdxCluster.getCrn());
+                    ThreadBasedUserCrnProvider.doAsInternalActor(() -> redbeamsServerEndpoint.updateClusterCrn(newSdxCluster.getEnvCrn(), newSdxCluster.getCrn(),
+                            sdxCluster.getCrn(), initiatorUserCrn));
```

Comment:
I'm confused about the naming here. The cluster is called `newSdxCluster`, but in the log message and in the call in `updateClusterCrn` it seems like `newSdxCluster.getCrn()` is the _existing_ cluster CRN, not the new one. What exactly is `newSdxCluster` here?

Reply:
I think this action's execution step has enough complexity to now warrant a unit test covering the different failure scenarios it might encounter

Reply:
I added comments and changed the name of the variables to provide more context so that it is easy to follow.

Reply:
The new comment says:
```
// We need to find the DatabaseServerConfig entry using the original CRN and update it to use the new CRN used by the detached data lake.
``
So if I'm understanding correctly, the database config being updated here is the config of the database attached to the original datalake. However, we gave that original datalake a new CRN so the resized datalake can use the original CRN. So the original database, which should still be attached to the original datalake until it's cleaned up, needs to be updated with original datalake's updated CRN.

If so I understand that. However, the naming is still bothering me. When we make the call, we're calling:
```
redbeamsServerEndpoint.updateClusterCrn(newResizedSdxCluster.getEnvCrn(), newResizedSdxCluster.getCrn(), detachingSdxCluster.getCrn(), initiatorUserCrn));
```
But the method definition is:
```
updateClusterCrn(String environmentCrn, String currentClusterCrn, String newClusterCrn, @InitiatorUserCrn String initiatorUserCrn)
```
So in the call, the second DL CRN is named "new", and in the method definition the first DL CRN is marked "new". It's a small thing, but it doesn't match and makes it confusing. Can you just rename it to `resizedSdxCluster` and just remove the conflict about the "new" variable name entirely?

Reply:
addressed